implementations/rust/ockam/ockam_command/src/project/ticket.rs

```diff
@@ -142,12 +156,16 @@ impl TicketCommand {
         // request an enrollment token that a future member can use to get a
         // credential.
         if let Some(id) = &self.member {
+            let attributes = self.attributes()?;
+            debug!("Attributes {:?}", attributes);
             authority_node_client
-                .add_member(ctx, id.clone(), self.attributes()?)
+                .add_member(ctx, id.clone(), attributes)
                 .await?
         } else {
+            let attributes = self.attributes()?;
+            debug!("Attributes {:?}", attributes);
```

Comment:
Since the attributes are used in both branches, we could extract this right before the `if let Some(...)`.

Also, I'd write the log line as `debug!(attributes = ?a);` or optionally set a message `debug!(attributes = ?a, "passed attributes");`.

- `%variable_name`: prints the `Display` version
- `?variable_name`: prints the `Debug` version

Reply:
great suggestion! good to know about the `%` and `?` meanings